iceoryx2/tests/sample_mut_tests.rs

```diff
@@ -0,0 +1,162 @@
+                .max_publishers(1)
+                .create::<u64>()
+                .unwrap();
+
+            let publisher = service
+                .publisher()
+                .max_loaned_samples(MAX_LOANED_SAMPLES)
+                .create()
+                .unwrap();
+
+            let subscriber = service.subscriber().create().unwrap();
+
+            Self {
+                service_name,
+                service,
+                publisher,
+                subscriber,
+            }
+        }
+    }
+
+    #[test]
+    fn when_going_out_of_scope_it_is_released<Sut: Service>() {
+        let test = TestFixture::<Sut>::new();
+
+        let mut sample_vec = vec![];
+
+        for _ in 0..4 {
+            while let Ok(sample) = test.publisher.loan() {
+                sample_vec.push(sample);
+            }
+
+            assert_that!(sample_vec, len MAX_LOANED_SAMPLES);
+            sample_vec.clear();
+        }
+    }
+
+    #[test]
+    fn header_tracks_correct_origin<Sut: Service>() {
+        let test = TestFixture::<Sut>::new();
+        let sample = test.publisher.loan().unwrap();
+        assert_that!(sample.header().publisher_id(), eq test.publisher.id());
+    }
+
+    #[test]
+    fn write_payload_works<Sut: Service>() {
+        const PAYLOAD_1: u64 = 891283689123555;
+        const PAYLOAD_2: u64 = 71820;
+        let test = TestFixture::<Sut>::new();
+        let sample = test.publisher.loan_uninit().unwrap();
+        let mut sample = sample.write_payload(PAYLOAD_1);
+
+        assert_that!(*sample.payload(), eq PAYLOAD_1);
+        assert_that!(*sample.payload_mut(), eq PAYLOAD_1);
+
+        *sample.payload_mut() = PAYLOAD_2;
+
+        assert_that!(*sample.payload(), eq PAYLOAD_2);
+        assert_that!(*sample.payload_mut(), eq PAYLOAD_2);
+    }
+
+    #[test]
+    fn assume_init_works<Sut: Service>() {
+        const PAYLOAD: u64 = 7182055123;
+        let test = TestFixture::<Sut>::new();
+        let mut sample = test.publisher.loan_uninit().unwrap();
+        let _ = *sample.payload_mut().write(PAYLOAD);
+        let mut sample = unsafe { sample.assume_init() };
+
+        assert_that!(*sample.payload(), eq PAYLOAD);
+        assert_that!(*sample.payload_mut(), eq PAYLOAD);
+    }
+
+    #[test]
+    fn send_works<Sut: Service>() {
+        const PAYLOAD: u64 = 3215357;
+        let test = TestFixture::<Sut>::new();
+        let sample = test.publisher.loan_uninit().unwrap();
+        let sample = sample.write_payload(PAYLOAD);
+
+        assert_that!(sample.send(), eq Ok(1));
+
+        let received_sample = test.subscriber.receive().unwrap().unwrap();
+        assert_that!(*received_sample, eq PAYLOAD);
+    }
+
+    #[test]
+    fn sample_of_dropped_service_does_block_new_service_creation<Sut: Service>() {
+        let test = TestFixture::<Sut>::new();
+        let service_name = test.service_name.clone();
+        let _sample = test.publisher.loan_uninit().unwrap();
+
+        drop(test);
```

Comment:
Food for thought. How about calling this `test_context`, `context` or `ctx`?

Reply:
Lets call it `TestContext`